rm-datasource/src/main/java/io/seata/rm/datasource/exec/BaseTransactionalExecutor.java

```diff
@@ -279,10 +278,12 @@ protected void prepareUndoLog(TableRecords beforeImage, TableRecords afterImage)
 
         TableRecords lockKeyRecords = sqlRecognizer.getSQLType() == SQLType.DELETE ? beforeImage : afterImage;
         String lockKeys = buildLockKey(lockKeyRecords);
-        connectionProxy.appendLockKey(lockKeys);
+        if (null != lockKeys) {
```

Comment:
the method return value ,  is from :  
`StringBuilder sb = new StringBuilder();
xxx
return sb.toString();`
It can not use null != lockkeys, should use if(StringUtils.isBlank(lockKeys))

Reply:
+1

Reply:
```
if (rowsIncludingPK.size() == 0) {
            return null;
        }
```